modules/core/src/main/java/org/apache/ignite/internal/jdbc2/lob/JdbcBlobBuffer.java

```diff
@@ -0,0 +1,354 @@
+    }
+
+    /**
+     * Create buffer which takes ownerhip of and wraps the existing byte array and starts working in
+     * the read-write mode.
+     *
+     * @param arr The byte array to be wrapped.
+     */
+    public static JdbcBlobBuffer createReadWrite(byte[] arr) {
+        return new JdbcBlobBuffer(new JdbcBlobMemoryStorage(arr));
+    }
+
+    /**
+     * Create empty buffer which starts working in the read-write mode.
+     */
+    public static JdbcBlobBuffer createReadWrite() {
+        return new JdbcBlobBuffer(new JdbcBlobMemoryStorage());
+    }
+
+    /**
+     * Create shallow copy of the buffer passed.
+     *
+     * <p>Sharing of the underlying storage is intended.
+     *
+     * @param other Other buffer.
+     */
+    public static JdbcBlobBuffer shallowCopy(JdbcBlobBuffer other) {
+        return new JdbcBlobBuffer(other.storage);
+    }
+
+    /**
+     * Create buffer which wraps the passed storage instance.
+     * @param storage Storage instance.
+     */
+    private JdbcBlobBuffer(JdbcBlobStorage storage) {
+        this.storage = storage;
+    }
+
+    /**
+     * @return Total number of bytes in the buffer.
+     */
+    public long totalCnt() {
+        return storage.totalCnt();
+    }
+
+    /**
+     * Provide OutputStream through which the data can be written to buffer starting from
+     * the (zero-based) position.
+     *
+     * @param pos The zero-based offset to the first byte to be written. Must not be negative
+     *            or greater than total count of bytes in buffer.
+     *
+     * @return OutputStream instance.
+     */
+    public OutputStream getOutputStream(long pos) {
+        long totalCnt = storage.totalCnt();
+
+        if (pos < 0 || pos > totalCnt)
+            throw new IndexOutOfBoundsException("Position cannot be negative or greater than total count of bytes in buffer " +
+                    "[pos=" + pos + ", totalCnt=" + totalCnt + "]");
+
+        return new BufferOutputStream(pos);
+    }
+
+    /**
+     * Provide InputStream through which the data can be read starting from the
+     * begining.
+     *
+     * <p>Stream is not limited meaning that it would return any new data
+     * written to the buffer after stream creation.
+     *
+     * @return InputStream instance.
+     */
+    public InputStream getInputStream() {
+        return new BufferInputStream();
+    }
+
+    /**
+     * Provide InputStream through which the no more than {@code len} bytes can be read
+     * from buffer starting from the specified zero-based position {@code pos}.
+     *
+     * @param pos The zero-based offset to the first byte to be retrieved. Must not be negative
+     *            or greater than total count of bytes in buffer.
+     * @param len The length in bytes of the data to be retrieved. Must not be negative.
+     * @return InputStream instance.
+     */
+    public InputStream getInputStream(long pos, long len) {
+        if (len < 0)
```

Comment:
Looks like sanity checks double. Here and in

`JdbcBlob#getBytes`, `JdbcBlob#getBinaryStream` where it's used.
Can we keep only one copy of sanity checks?